uint8 trigger
int32 seq_no